packages/mobile/src/walletConnect/request.ts

```diff
@@ -29,14 +36,56 @@ export function* handleRequest({ method, params }: { method: string; params: any
   yield call(unlockAccount, account)
 
   switch (method) {
-    case SupportedActions.eth_signTransaction:
-      return (yield call(wallet.signTransaction.bind(wallet), params[0])) as EncodedTransaction
+    case SupportedActions.eth_signTransaction: {
+      // IMPORTANT: We need to normalize the transaction parameters
+      // WalletConnect v1 utils currently strips away important fields like `chainId`, `feeCurrency`, `gatewayFee` and `gatewayFeeRecipient`
+      // See https://github.com/WalletConnect/walletconnect-monorepo/blame/c6b26481c34848dbc9c49bb0d024bda907ec4599/packages/helpers/utils/src/ethereum.ts#L66-L86
+      // Also the dapp developer may have omitted some of the needed fields,
+      // so it's nice to be flexible and still allow the transaction to be signed (and sent) successfully
+
+      const rawTx = { ...params[0] }
+      let tx
+      // Provide an escape hatch for dapp developers who don't want any normalization
+      if (rawTx.__skip_normalization) {
```

Comment:
Will WC strip this field out though?

Reply:
Yes it will for v1 requests, so developers wanting to use it would have to both set `type = null` and `__skip_normalization`.